tracer/src/Datadog.Trace.SourceGenerators/PublicApi/PublicApiGenerator.cs

```diff
@@ -39,16 +39,19 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                         GeneratePublicApiAttribute,
                         static (node, _) => node is PropertyDeclarationSyntax,
                         static (context, ct) => GetPublicApiProperties(context, ct))
-                   .Where(static m => m is not null)!;
+                   .Where(static m => m is not null)!
```

Comment:
Just curious: what is the `!` for here? Even in the previous version, it doesn't make sense to me. Can `Where()` ever return `null`?

Reply:
I don't think the `!` is required